packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx

```diff
@@ -217,9 +217,8 @@ function ChartsXAxis(inProps: ChartsXAxisProps) {
         const xTickLabel = labelOffset ?? 0;
         const yTickLabel = positionSign * (tickSize + 3);
 
-        const showTick = offset >= left - 1 && offset <= left + width + 1;
-        const showTickLabel =
-          offset + xTickLabel >= left - 1 && offset + xTickLabel <= left + width + 1;
+        const showTick = isPointInside({ x: offset }, { direction: 'x' });
+        const showTickLabel = isPointInside({ x: offset + xTickLabel }, { direction: 'x' });
```

Comment:
I'm not sure if it matters much, but the shape `{ x?: number, y:? number }` is [un-monomorphizing](https://romgrk.com/posts/optimizing-javascript#2-avoid-different-shapes) the `isPointInside` function. If it's a hot function, could be worth to pass a placeholder value like `{ x: offset, y: -1 }`.